### PR TITLE
feat(component): ProgressBar

### LINKED ISSUE #65: Progress Bar
Implement progress bar in Shelley:
- https://react-spectrum.adobe.com/react-aria/useProgressBar.html

Spectrum example:
https://react-spectrum.adobe.com/react-spectrum/ProgressBar.html

### PR DESCRIPTION
**Pull request by cristinallamas**
resolves #65

### CODE REVIEW

File: src/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    </div>
+  );
+}
+
+/**
+ * ProgressBars show the progression of a system operation such as downloading, uploading, or processing, in a visual way.
+ * They can represent determinate or indeterminate progress.
+ */
+const ForwardedProgressBar = forwardRef(ProgressBar);
+export { ForwardedProgressBar as ProgressBar };
```
Comment (mnapthine) [inline src/ProgressBar/ProgressBar.tsx:145]: We might need to add something to the `vscode` config to auto add end of file line breaks on-save.

Comment (cristinallamas) [reply src/ProgressBar/ProgressBar.tsx:147]: YES!

File: cypress/component/ProgressBar.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    cy.get(progressBar)
+      .should("have.attr", "class")
+      .and("to.have.string", "ProgressBar")
+      .should("to.have.string", "multistep")
+
+    cy.get('[data-id="stepIndicator"]').should("have.length", multiSteProps["totalSteps"]);
+    // cy.get('[data-id="stepIndicatorFill"]').eq(multiSteProps["currentStep"]).should("have.attr", "class")
+    //   .and("to.have.string", "isActive")
+    // cy.get('[data-id="stepIndicatorFill"]').eq(multiSteProps["currentStep"]).should("have.css", "width", `${multiSteProps["stepProgress"]}%`);
```
Comment (balinttamas87) [inline cypress/component/ProgressBar.cy.tsx:94]: Is this still needed?

File: cypress/component/ProgressBar.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    );
+
+    cy.get(progressBar)
+      .should("have.attr", "class")
+      .and("to.have.string", "ProgressBar")
+      .and("to.have.string", "cypress-test");
+  });
+
+  it("forwards ref", () => {
+    const ref = React.createRef<HTMLDivElement>();
```
Comment (balinttamas87) [inline cypress/component/ProgressBar.cy.tsx:33]: Just a note: These 2 tests could probably be reused for most of the components.

File: src/ProgressBar/ProgressBar.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                index={index}
+                currentStep={currentStep}
+                stepProgress={stepProgress}
+              />
+            ))}
+          </div>
+        ) : (
+          <div className={st(classes.stepIndicator)}>
+              <div
+                style={{ width: barWidth, height: 10 }}
```
Comment (balinttamas87) [inline src/ProgressBar/ProgressBar.tsx:128]: Would it be useful for `height: 10` to be part of the CSS and customisable/overridable?

File: src/ProgressBar/ProgressBar.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    stepProgress: number;
+  }) => {
+
+    const fillStyle: CSSProperties = {
+      width: index === currentStep - 1 ? `${stepProgress}%` : "100%",
+      height: "100%",
+    };
+    const currentStepClassName = st(classes.stepIndicatorFill, {
+      isActive: index < currentStep ? true : false,
+    });
```
Comment (balinttamas87) [inline src/ProgressBar/ProgressBar.tsx:94]: nit: Could we have a line break between these two declarations?

**Review by balinttamas87 — COMMENTED** [review]
Great PR @cristinallamas . Left a couple of comments. Nothing major really. Let me know what you think.

File: cypress/component/ProgressBar.cy.tsx (hunk 2 above)
Comment (cristinallamas) [reply cypress/component/ProgressBar.cy.tsx:94]: currently fixing this part for it to work as I think it's good to have it

File: cypress/component/ProgressBar.cy.tsx (hunk 3 above)
Comment (cristinallamas) [reply cypress/component/ProgressBar.cy.tsx:33]: I've copied it from ProgressCircle, so I'll leave it separately for now

File: cypress/component/ProgressBar.cy.tsx [hunk 6]
```diff
@@ -88,13 +89,18 @@ describe("ProgressBar", () => {
...
       .should("to.have.string", "multistep")
 
-    cy.get('[data-id="stepIndicator"]').should("have.length", multiSteProps["totalSteps"]);
-    // cy.get('[data-id="stepIndicatorFill"]').eq(multiSteProps["currentStep"]).should("have.attr", "class")
-    //   .and("to.have.string", "isActive")
-    // cy.get('[data-id="stepIndicatorFill"]').eq(multiSteProps["currentStep"]).should("have.css", "width", `${multiSteProps["stepProgress"]}%`);
-  });
-
+    cy.get(stepIndicator)
+      .should("have.length", multiSteProps["totalSteps"]);
```
Comment (balinttamas87) [inline cypress/component/ProgressBar.cy.tsx:93]: Did you mean `multiStepProps` with 2 `p`s?

**Review by balinttamas87 — APPROVED** [review]

File: cypress/component/ProgressBar.cy.tsx (hunk 6 above)
Comment (cristinallamas) [reply cypress/component/ProgressBar.cy.tsx:93]: saving in letters? 😆 fixing it

**Review by mnapthine — APPROVED** [review]

**Pull request merged**